core/src/impl/Kokkos_Default_GraphNodeKernel.hpp

```diff
@@ -63,7 +63,7 @@ template <class ExecutionSpace>
 struct GraphNodeKernelDefaultImpl {
   // TODO @graphs decide if this should use vtable or intrusive erasure via
   //      function pointers like in the rest of the graph interface
-  virtual void execute_kernel() const = 0;
```

Comment:
why does the constness change here and below?

Reply:
Because `Impl::ParallelReduce<>::execute()` isn't `const` for the `Cuda` specialization, and we need to call that in `Cuda` versions before 10.0.

Reply:
We looked at this a bit more. There are deeper issues here, and we may change the const ness back if we can resolve the deeper issues. Largely it has to do with the buffers for reductions which need to potentially grow, and the pointers which are part of the driver and hence have to change during execute. There are issues here with respect to multiple streams, mixing of delayed graph submit and non-graph dispatch, as well as multi threaded dispatch with the non-graph stuff. But we should go ahead with this PR and solve those corner cases separately.